pkg/dockerregistry/server/auth.go

```diff
@@ -151,15 +137,15 @@ func (ac *AccessController) Authorized(ctx context.Context, accessRecords ...reg
 				verb = "get"
 			default:
 				challenge.err = fmt.Errorf("Unknown action: %s", access.Action)
+				log.Error(challenge.err)
 				return nil, challenge
 			}
 
-			if err := verifyImageStreamAccess(repoParts[0], repoParts[1], verb, client); err != nil {
+			if err := verifyImageStreamAccess(imageStreamNS, imageStreamName, verb, client); err != nil {
 				challenge.err = err
 				return nil, challenge
 			}
 
-			return WithUserClient(ctx, client), nil
```

Comment:
moved this to the end of the function, to continue looping through all access records